redis-om-spring/src/main/java/com/redis/om/spring/client/RedisModulesClient.java

```diff
@@ -1,38 +1,82 @@
 package com.redis.om.spring.client;
 
 import org.springframework.data.redis.connection.jedis.JedisConnectionFactory;
-
+import redis.clients.jedis.Jedis;
+import redis.clients.jedis.JedisCluster;
+import redis.clients.jedis.JedisPooled;
+import redis.clients.jedis.UnifiedJedis;
+import redis.clients.jedis.bloom.commands.BloomFilterCommands;
+import redis.clients.jedis.bloom.commands.CountMinSketchCommands;
+import redis.clients.jedis.bloom.commands.CuckooFilterCommands;
+import redis.clients.jedis.bloom.commands.TopKFilterCommands;
+import redis.clients.jedis.json.RedisJsonCommands;
+import redis.clients.jedis.search.RediSearchCommands;
 import com.google.gson.GsonBuilder;
-import com.redislabs.modules.rejson.JReJSON;
 
-import redis.clients.jedis.Jedis;
+import java.util.Objects;
+import java.util.Optional;
 
 public class RedisModulesClient {
 
-  private GsonBuilder builder;
+  private final GsonBuilder builder;
+  private final UnifiedJedis unifiedJedis;
 
   public RedisModulesClient(JedisConnectionFactory jedisConnectionFactory, GsonBuilder builder) {
     this.jedisConnectionFactory = jedisConnectionFactory;
     this.builder = builder;
+    this.unifiedJedis = getUnifiedJedis();
+  }
+
+  public RedisJsonCommands clientForJSON() {
+    return unifiedJedis;
+  }
+
+  public RediSearchCommands clientForSearch() {
+    return unifiedJedis;
   }
 
-  public JReJSON clientForJSON() {
-    JReJSON client = new JReJSON(getJedis());
-    client.setGsonBuilder(builder);
-    return client;
+  public BloomFilterCommands clientForBloom() {
+    return unifiedJedis;
   }
 
-  public io.redisearch.Client clientForSearch(String index) {
-    return new io.redisearch.client.Client(index, getJedis());
+  public CountMinSketchCommands clientForCMS() {
+    return unifiedJedis;
   }
 
-  public io.rebloom.client.Client clientForBloom() {
-    return new io.rebloom.client.Client(getJedis());
+  public CuckooFilterCommands clientForCuckoo() {
+    return unifiedJedis;
+  }
+
+  public TopKFilterCommands clientForTopK() {
+    return unifiedJedis;
+  }
+
+  public UnifiedJedis getUnifiedJedis() {
```

Comment:
I think we should we define this method as private? WDYT?

Reply:
refactored!